app/models/course.rb

```diff
@@ -53,6 +53,12 @@ def generate_registration_key
     self.registration_key = 'C'.freeze + SecureRandom.base64(8)
   end
 
+  # Returns the root folder of the course.
+  # @return [Course::Material::Folder] The root folder.
+  def root_folder
+    material_folders.find_by(parent: nil)
```

Comment:
that's a problem with the course if we don't have this right? Otherwise the nil will probagate downwards.